File: src/com/mapdigit/game/box2d/director/CollisionDirector.scala
package com.mapdigit.game.box2d.director

import com.guidebee.game.GameEngine._
import com.guidebee.game.audio.Sound
import com.guidebee.game.scene.Actor

import com.guidebee.game.scene.collision.{Collision, CollisionListener}


/**
 * Created by james on 7/10/2014.
 */
object CollisionDirector extends CollisionListener  {
  private val dropSound = assetManager.get("drop.wav", classOf[Sound])

  override def CollisionDetected(collision: Collision): Unit = {
    if(collision!=null) {
      val actor1 = collision.getObjectA().asInstanceOf[Actor]
      val actor2 = collision.getObjectB().asInstanceOf[Actor]

      if (actor1 != null && actor2 != null) {

        if(actor1.getName=="Bucket" && actor2.getName=="RainDrop"){
          actor2.getParent.removeActor(actor2)
          dropSound.play()
        }else if(actor2.getName=="Bucket" && actor1.getName=="RainDrop"){
          actor1.getParent.removeActor(actor1)
          dropSound.play()
        }

      }
    }
  }
}

File: src/com/mapdigit/game/tutorial/drop/director/CollisionDirector.java
package com.mapdigit.game.tutorial.drop.director;

import com.guidebee.game.Collidable;
import com.guidebee.game.audio.Sound;
import com.guidebee.game.scene.Actor;
import com.guidebee.game.scene.collision.Collision;
import com.guidebee.game.scene.collision.CollisionListener;
import com.mapdigit.game.tutorial.drop.actor.Mario;
import com.mapdigit.game.tutorial.drop.actor.RainDrop;

import static com.guidebee.game.GameEngine.assetManager;


public class CollisionDirector implements CollisionListener {

    private Sound dropSound=assetManager.get("drop.wav",Sound.class);

    @Override
    public void collisionDetected(Collision collision) {

        if(collision!=null){
            Collidable objectA=collision.getObjectA();
            Collidable objectB=collision.getObjectB();
            if(objectA instanceof Mario){
                if(objectB instanceof RainDrop){
                    ((Actor)objectB).getParent().removeActor((Actor) objectB);
                    dropSound.play();
                }
            }else if(objectA instanceof RainDrop){
                if(objectB instanceof Mario){
                    ((Actor)objectA).getParent().removeActor((Actor) objectA);
                    dropSound.play();
                }
            }

        }


    }
}
